daemons/compute/server/servers/server_default.go

```diff
@@ -247,7 +256,136 @@ func (s *defaultServer) Create(ctx context.Context, req *pb.DataMovementCreateRe
 		return nil, err
 	}
 
-	source, err := s.findRabbitRelativeSource(ctx, req)
+	computeClientMount, computeMountInfo, err := s.findComputeMountInfo(ctx, req)
+	if err != nil {
+		return &pb.DataMovementCreateResponse{
+			Status:  pb.DataMovementCreateResponse_FAILED,
+			Message: err.Error(),
+		}, nil
+	}
+
+	response := &pb.DataMovementCreateResponse{}
+
+	if computeMountInfo.Type == "lustre" {
+		response, err = s.createNnfDataMovement(ctx, req, userId, groupId, computeClientMount)
+	} else {
+		response, err = s.createRsyncNodeDataMovement(ctx, req, userId, groupId, computeMountInfo)
+	}
+	return response, err
+}
+
+func getDirectiveIndexFromClientMount(object *dwsv1alpha1.ClientMount) (string, error) {
+	// Find the DW index for our work.
+	labels := object.GetLabels()
+	if labels == nil {
+		return "", fmt.Errorf("Unable to find labels on compute ClientMount, namespaces=%s, name=%s", object.Namespace, object.Name)
+	}
+
+	dwIndex, found := labels[nnfv1alpha1.DirectiveIndexLabel]
+	if !found {
+		return "", fmt.Errorf("Unable to find directive index label on compute ClientMount, namespace=%s name=%s", object.Namespace, object.Name)
+	}
+
+	return dwIndex, nil
+}
+
+func (s *defaultServer) createNnfDataMovement(ctx context.Context, req *pb.DataMovementCreateRequest, userId uint32, groupId uint32, computeClientMount *dwsv1alpha1.ClientMount) (*pb.DataMovementCreateResponse, error) {
+
+	var dwIndex string
+	if dw, err := getDirectiveIndexFromClientMount(computeClientMount); err != nil {
+		return &pb.DataMovementCreateResponse{
+			Status:  pb.DataMovementCreateResponse_FAILED,
+			Message: err.Error(),
+		}, nil
+	} else {
+		dwIndex = dw
+	}
+
+	lustrefs, err := s.findDestinationLustreFilesystem(ctx, req.GetDestination())
+	if err != nil {
+		return &pb.DataMovementCreateResponse{
+			Status:  pb.DataMovementCreateResponse_FAILED,
+			Message: err.Error(),
+		}, nil
+	}
+
+	// We don't have the actual NnfDataMovement parent available, but we know the name
+	// and the namespace because they will match the workflow's name and namespace.
+	parentDm := &nnfv1alpha1.NnfDataMovement{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      req.Workflow,
+			Namespace: req.Namespace,
+		},
+	}
+
+	// We don't have the actual source NnfStorage available, but we know
```

Comment:
This seems error prone if we ever want to change the name of nnfstorage; I thought it was part of the client mount?

Reply:
For lustre-to-lustre, the ClientMount that is created during  pre_run by nnf_access_controller is this:
```
spec:
  desiredState: mounted
  mounts:
  - device:
      lustre:
        fileSystemName: zaph152
        mgsAddresses:
        - 172.31.241.111@tcp
      type: lustre
    mountPath: /mnt/nnf/20152/job/zaph152
    options: ""
    targetType: directory
    type: lustre
  node: rabbit-dev-cn-01
status:
```

@matthew-richerson, the NnfAccess that is creating this Clientmount has the NnfStorage object ref.  Do  you see any reason I couldn't add an ObjectReference for it into ClientMountDeviceLustre to make it easily available to  the nnf-dm daemon?